core/src/input_manager.cpp

```diff
@@ -0,0 +1,205 @@
+#include <GLFW/glfw3.h>
+#include <knoting/input_manager.h>
+#include <knoting/window.h>
+#include <algorithm>
+
+namespace knot {
+
+input_manager::input_manager(Window& owner) : m_sensitivity(0.3f), m_lastScroll(0.0f) {
+    for (int i = 0; i < (int)KeyCode::Last; ++i) {
+        m_keyBinding[i] = false;
+        m_holdKeyBindings[i] = false;
+    }
+
+    for (int i = 0; i < (int)MouseButtonCode::Last; ++i) {
+        m_mouseBindings[i] = false;
+        m_holdMouseBindings[i] = false;
+    }
+
+    for (int i = 0; i < (int)PadCode::Last; ++i) {
+        for (int j = 0; j < (int)PadButtonCode::Last; ++j) {
+            m_padBindings[i][j] = false;
+            m_holdKeyBindings[i] = false;
+        }
+
+        for (int j = 0; j < (int)JoyStickCode::Last; ++j) {
+            m_joyStickBindings[i][j] = 0.0f;
+        }
+    }
+}
+
+input_manager::~input_manager() {}
+
+/// Start Key Keyboard
+[[maybe_unused]] bool input_manager::m_keyPressed(KeyCode key) {
```

Comment:
`[[maybe_unused]]` not needed, ignore the warnings, they will disappear naturally as we start using them.